src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php

```diff
@@ -137,28 +147,45 @@ public function dispatch($eventName, Event $event = null)
         $this->preDispatch($eventName, $event);
 
         $e = $this->stopwatch->start($eventName, 'section');
+        $previousRequestHash = $this->currentRequestHash;
 
-        $this->dispatcher->dispatch($eventName, $event);
+        try {
+            $this->dispatcher->dispatch($eventName, $event);
+            $this->currentRequestHash = $previousRequestHash;
 
-        if ($e->isStarted()) {
-            $e->stop();
-        }
+            if ($e->isStarted()) {
+                $e->stop();
+            }
+
+            $this->postDispatch($eventName, $event);
+        } finally {
```

Comment:
make sure post process is called after e.g. kernel.exception, or any exception for that matter.